http-clients/url-connection-client/src/main/java/software/amazon/awssdk/http/urlconnection/UrlConnectionHttpClient.java

```diff
@@ -140,6 +141,9 @@ private HttpURLConnection createAndConfigureConnection(HttpExecuteRequest reques
         // See: https://github.com/aws/aws-sdk-java-v2/issues/975
         connection.setInstanceFollowRedirects(false);
 
+        request.httpRequest().firstMatchingHeader(CONTENT_LENGTH).map(Long::parseLong)
```

Comment:
Do we need `.map(Long::parseLong)`?

Reply:
Yes, pretty sure we do - [`firstMatchingHeader(CONTENT_LENGTH)`](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/http/SdkHttpHeaders.html#firstMatchingHeader-java.lang.String-) returns an `Optional<String>`, and [`setFixedLengthStreamingMode()`](https://docs.oracle.com/javase/8/docs/api/java/net/HttpURLConnection.html#setFixedLengthStreamingMode-long-) accepts a `long`, so we need to map that `String` into a `long`.

Reply:
Ah yeah, got it